File: Euclid/Game/Navigator/NavigatorCategory.cs
using Euclid.Storage.Database.Data;

namespace Euclid.Game
{
    public class NavigatorCategory
    {
        #region Fields

        private readonly NavigatorCategoryData categoryData;

        #endregion

        #region Properties

        public NavigatorCategoryData Data
        {
            get { return categoryData; }
        }

        #endregion

        #region Constructors

        public NavigatorCategory(NavigatorCategoryData categoryData)
        {
            this.categoryData = categoryData;
        }

        #endregion

        #region Public methods

        public int GetCurrentVisitors()
        {
            int currentVisitors = 0;

            foreach (Room room in RoomManager.Instance.Rooms.Values)
            {
                if (room.Data.CategoryId == this.categoryData.Id)
                {
                    currentVisitors += room.Data.UsersNow;
                }
            }

            return currentVisitors;
        }

        public int GetMaxVisitors()
        {
            int maxVisitors = 0;

            foreach (Room room in RoomManager.Instance.Rooms.Values)
            {
                if (room.Data.CategoryId == this.categoryData.Id)
                {
                    maxVisitors += room.Data.UsersMax;
                }
            }

            return maxVisitors;
        }

        #endregion
    }
}

File: euclid/src/main/java/euclid/game/navigator/NavigatorCategory.java
package euclid.game.navigator;

import euclid.game.room.Room;
import euclid.game.room.RoomManager;
import euclid.storage.database.data.NavigatorCategoryData;

public class NavigatorCategory {

    private final NavigatorCategoryData categoryData;

    public NavigatorCategory(NavigatorCategoryData categoryData) {
        this.categoryData = categoryData;
    }

    public NavigatorCategoryData getData() {
        return categoryData;
    }

    public int getCurrentVisitors() {
        int currentVisitors = 0;

        for (Room room : RoomManager.getInstance().getRooms().values()) {
            if (room.getData().getCategoryId() == this.categoryData.getId()) {
                currentVisitors += room.getData().getUsersNow();
            }
        }

        return currentVisitors;
    }

    public int getMaxVisitors() {
        int maxVisitors = 0;

        for (Room room : RoomManager.getInstance().getRooms().values()) {
            if (room.getData().getCategoryId() == this.categoryData.getId()) {
                maxVisitors += room.getData().getUsersMax();
            }
        }

        return maxVisitors;
    }
}
